core/src/main/java/com/radixdlt/RunningRadixNode.java

```diff
@@ -118,7 +124,11 @@ public static RunningRadixNode run(UnstartedRadixNode unstartedRadixNode) {
 
     for (var module : moduleStartOrder) {
       final var moduleRunner = moduleRunners.get(module);
-      moduleRunner.start();
+      moduleRunner.start(
+          error -> {
+            log.error("Unhandled exception in runner {}; shutting down the node", module, error);
+            runningNode.shutdown();
```

Comment:
Hmm this no longer actually exits the process? But potentially leaves the main thread around doing nothing?

Perhaps we should actually do an exit in the `shutdown()` method? Or else just trigger `Runtime.shutdown()` in there or something? (which I imagine would then run the shutdown hook?

Reply:
> Hmm this no longer actually exits the process? 

This exits the process (I tested it from IntelliJ and terminal in some local experiments), by ensuring that all threads started (by `RunningRadixNode.run()`) are stopped (by `RunningRadixNode.shutdown()`), which is a correct way of spinning down a service (contrary to `System.exit()`, which was causing us problems).

> But potentially leaves the main thread around doing nothing?

The main thread is probably long dead when all this happens.
(because `bootstrapRadixNode()` is effectively async - it starts a few threads belonging to the Node and returns)

Reply:
> This exits the process (I tested it from IntelliJ and terminal in some local experiments), by ensuring that all threads started (by RunningRadixNode.run()) are stopped (by RunningRadixNode.shutdown()), which is a correct way of spinning down a service (contrary to System.exit(), which was causing us problems).

But this assumes that we have properly shut down all threads. Otherwise if we introduce a regression in future, we could have a zombie node which doesn't restart automatically, which seems scary to me?

It feels weird that we use the same method for "on shutdown" hook as we do for "please shutdown" - I feel like we should instead tell the JVM "please shut me down", which will then try to gracefully shut down - but also fall back to  non-graceful shutdown if it has to - (which I believe is `System.Exit()`?)

Reply:
hmm, a `System.exit()` called by the "on exception" business callback was the cause of other problems, so I wanted to avoid it...

But I totally see what you want to ensure. Is it feasible to have an integration test that starts an actual process (rigged so that some fatal exception happens) and asserts on the process terminating itself? Would it make this PR less scary?
_(TBH a positive test counterpart [i.e. assert the process starts and stabilizes fine] would also be nice)_

But if not feasible (or the required infra would not fit in this PR, or within priorities), then I guess I can switch to the "request System.exit() and spin yourself down in a shutdown hook" approach.

Reply:
Alright, I realized it is very much ok to go this way, since `RunningRadixNode` is not used by any deterministic/simulation int-tests (only by prod).

So, we no longer risk `... finished with non-zero exit value 255` from any tests (in the most recent commit I also remove this warnings from comments).

PTAL.

Reply:
Also happy with creating an integration test to ensure the shutdown hook does shut down all threads, but yes - this is only used in prod

Reply:
Do the tests also use the shutdown hook? Or do their own thing?

Reply:
> integration test to ensure the shutdown hook does shut down all threads

Actually not shutdown hook, but JVM is supposed to halt all threads after all shutdown hooks run. I'm gonna trust the JVM :p

(please note how JVM can still hang if e.g. some shutdown hook does NOT return :D ahh the good old Halting Problem)

> Do the tests also use the shutdown hook? Or do their own thing?

Own thing: 
https://github.com/radixdlt/babylon-node/blob/df358cd19debff098855e8a031828ed1f678b7a3/core/src/test-core/java/com/radixdlt/harness/simulation/network/SimulationNodes.java#L262-L263
(that's why I could fulfill your wish without breaking the elegant behavior of tests)